NewsTests/Sceen/Main/NewsViewModelTest.swift

```diff
@@ -0,0 +1,57 @@
+//
+//  NewsViewModelTest.swift
+//  NewsTests
+//
+//  Created by Mohamed Elboraey on 18/05/2025.
+//
+
+import XCTest
+@testable import News
+
+final class NewsViewModelTest: XCTestCase {
+    
+
+    let viewModel = NewsViewModel()
+
+    override func setUpWithError() throws {
+        // Put setup code here. This method is called before the invocation of each test method in the class.
+    }
+
+    override func tearDownWithError() throws {
+        // Put teardown code here. This method is called after the invocation of each test method in the class.
+    }
+
+    func test_newsData_whenUrlIsValidAndThereAreData_shouldBeChangedWithNewData() async {
+//        Given
+        let url = "https://gnews.io/api/v4/top-headlines?category=general&lang=en&country=us&max=10&apikey=b9c748f3a9fcb0e9f7b8d6b2d86c17cc"
+//        When
+        await viewModel.getData(from: url)
+        let newData = viewModel.newsData
+//        Then
+        XCTAssertNotNil(newData)
+        
+    }
```

Comment:
```suggestion
    func test_newsData_whenUrlIsValidAndThereAreData_shouldBeChangedWithNewData() async {
        // Given
        let url = "https://gnews.io/api/v4/top-headlines?category=general&lang=en&country=us&max=10&apikey=b9c748f3a9fcb0e9f7b8d6b2d86c17cc"

        // When
        await viewModel.getData(from: url)
        let newData = viewModel.newsData

        // Then
        XCTAssertNotNil(newData)
    }
```
nit - Indentation and linting